src/rustzcash.rs

```diff
@@ -77,6 +78,33 @@ fn read_le(from: &[u8]) -> FrRepr {
     f
 }
 
+/// Reads an FsRepr from [u8] of length 32
+/// This will panic (abort) if length provided is
+/// not correct
+fn read_fs(from: &[u8]) -> FsRepr {
```

Comment:
This will read from bytes into an Fs in little-endian bit order, which is what we want, but the function `librustzcash_crh_ivk` produces a little-endian _byte_ order result and writes that out. Then, when you call `librustzcash_ivk_to_pkd`, it's not going to work correctly.

Reply:
Does it? I thought the intention was that the output of `BLAKE2s` was exactly the representation we wanted. More precisely, I think this is fine, because:

- Here we are reading a little-endian byte array in big-endian, and *then* flipping, which means we need to flip both the u64s and their bits.
- In `ViewingKey.ivk()` we flip the little-endian byte array *before* reading it as big endian, and since the endianness of the bits within each byte doesn't matter for reading, that should mean the result is the same as flipping the bits above.

I'll try and generate some test vectors to confirm.

Reply:
Now that #12 has been merged, this function needs to be updated (to use `read_le` and not swap bits).